.github/workflows/integration.yaml

```diff
@@ -104,13 +104,247 @@ jobs:
           RUST_LOG: info
           RUST_BACKTRACE: 1
           NO_COLOR_LOG: 1
-          CHAIN_COMMAND_PATH: ${{ matrix.chain.command }}
-          ACCOUNT_PREFIX: ${{ matrix.chain.account_prefix }}
+          CHAIN_COMMAND_PATHS: ${{ matrix.chain.command }}
+          ACCOUNT_PREFIXES: ${{ matrix.chain.account_prefix }}
         with:
           max_attempts: 3
           timeout_minutes: 45
           command: |
-            nix shell .#python .#${{ matrix.chain.package }} -c cargo \
+            nix shell .#python ${{ matrix.chain.package }} -c cargo \
+              test -p ibc-integration-test --no-fail-fast -- \
+              --nocapture --test-threads=2
+
+  multi-chains-test:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        package1: [.#gaia6, .#gaia7, .#gaia8, .#ibc-go-v2-simapp, .#ibc-go-v3-simapp, .#ibc-go-v4-simapp, .#ibc-go-v5-simapp, .#ibc-go-v6-simapp, .#wasmd, .#evmos, .#osmosis]
+        package2: [.#gaia6, .#gaia7, .#gaia8, .#ibc-go-v2-simapp, .#ibc-go-v3-simapp, .#ibc-go-v4-simapp, .#ibc-go-v5-simapp, .#ibc-go-v6-simapp, .#wasmd, .#evmos, .#osmosis]
```

Comment:
I think we could make it a matrix of dictionaries. Something like:

```
matrix:
  first-package:
    - package: gaia6
      chain_command_path: gaiad
      account_prefix: cosmos
```

Then we don't need the `exclude` list below.

Btw I am not quite sure why we need to specify the `.#` prefix everywhere here. It might be simpler to write it as `nix shell .#${{ matrix.first_package.package }}`.